docs/changes/3.0.0.rst

```diff
@@ -14,6 +14,57 @@ Changes to Supported Platforms
 
 .. towncrier release notes start
 
+
+
+
+Breaking Changes
+----------------
+
+- Starting with Falcon 3.0, Python 2 is no longer supported. Therefore, support
+  for CPython 2.7 and PyPy2.7 was removed in this release. We will continue to
+  maintain the Falcon 2.x series for those who still need to run under Python 2,
+  at least until that platform's scheduled EOL (January 1, 2020 at the time of
+  this writing; see also PEP 373). (27-support)
+- Python 3.4 has now reached its end-of-life per PEP 429, and as a result is no
+  longer supported as of Falcon 3.0. (34-support)
+- The deprecated ``stream_len`` property was removed from the ``Response`` class.
+  Please use ``Response.set_stream()`` or ``Response.content_length`` instead. (stream_len)
+
+
+New & Improved
+--------------
+
+- A new kwarg, ``cors_enable``, was added to the :class:`falcon.API` initializer.
+  ``cors_enable`` can be used to enable a simple blanket CORS policy for all
+  responses. (See also: :ref:`cors`.) (1194-cors)
+- Falcon now supports all WebDAV methods (RFC 2518 and RFC 4918), such as COPY, LOCK, MKCOL, MOVE, PROPFIND, PROPPATCH and UNLOCK. (1426-webdav)
+- The :meth:`~.Response.set_cookie` method now supports setting the
+  `SameSite` cookie attribute. (1557-cookie-samesite)
+- The :meth:`~.Response.add_link` method now supports setting the `crossorigin`
+  link CORS settings attribute. (1574-add-link-crossorigin)
+- The :attr:`~.RequestOptions.auto_parse_form_urlencoded` option is now
+  deprecated in favor of :class:`falcon.media.URLEncodedFormHandler`.
+  (See also: :ref:`access_urlencoded_form`). (1580-auto-parse-form-urlencoded)
+- Exceptions are now handled by the registered handler for the most specific matching exception class, rather than in reverse order of registration. "Specificity" is determined by the method resolution order of the raised exception type. (See :meth:`~.App.add_error_handler` for more details.) (1603-smart-exception-handler-precedence)
+- :meth:`~.Request.get_param_as_bool` now supports the use of ``'t'`` and ``'y'``
+  values for ``True``, as well as ``'f'`` and ``'n'`` for ``False``. (support-additional-truthy-param-values)
+- ``URLEncodedFormHandler`` was added for handling URL-encoded forms (of content
+  type ``application/x-www-form-urlencoded``) as ``Request.media``. (urlencoded-form-media-handler)
+
+
+Fixed
+-----
+
+- Header-related methods of the ``falcon.Response`` class no longer coerce the
+  passed header name to a string via ``str()``. (response-header-coercion)
+
+
+Misc
+----
+
+- Setup towncrier to make CHANGES reporting much easier. (use-towncrier)
+
+
```

Comment:
These changes are not to be merged. I included them to demonstrate what the output looks like.